src/runcrate/converters/base.py

```diff
@@ -0,0 +1,129 @@
+from rocrate.model.contextentity import ContextEntity
+
+from ..constants import PROFILES_BASE, PROFILES_VERSION, WROC_PROFILE_VERSION
+
+
+class converter:
+    def __init__(self):
+        self.root = None
+        self.workflow_name = None
+        self.license = None
+        self.readme = None
+        self.wf_path = None
+        self.workflow_definition = {}
+        self.step_maps = {}
+        self.ro = None
+        self.with_prov = set()
+        self.workflow_run = None
+        self.roc_engine_run = None
+        self.control_actions = {}
+        self.collections = {}
+        self.hashes = {}
+        self.file_map = {}
+        self.manifest = None
+
+    # --------------------------------------------------------------------------
+    # Top level functions - called by the build() function
+
+    def add_root_metadata(self, crate):
```

Comment:
There are only two common functions in the base class, not enough to justify the base converter + specialized converter layout

Reply:
Is there an alternative that jumps out to you? It seemed the most natural fit to me

Overall I can see where you are coming from but I think there are a few points in favour of the approach I took.

I think the top level functions are common they just have different implementations
I think more stuff is likely to move into base as my understanding improves / use cases accumulate
Base sets the template and highlights when a method has not been implemented (by raising)